geom/geom/inc/TGeoNavigator.h

```diff
@@ -199,6 +203,35 @@ class TGeoNavigator : public TObject {
       fLastPoint[1] = y, fLastPoint[2] = z;
    }
 
+   // Check if we have a cached safety value from parallel world, and if this can still be used.
+   // Return negative value if no cache available.
+   Double_t GetPWSafetyEstimateFromCache(Double_t cpoint[3]) const
+   {
+      // disregard too small or invalid safeties
+      if (fLastPWSafety < TGeoShape::Tolerance()) {
+         return -1.;
+      }
+      const auto d0 = fLastPWSaftyPnt[0] - cpoint[0];
+      const auto d1 = fLastPWSaftyPnt[1] - cpoint[1];
+      const auto d2 = fLastPWSaftyPnt[2] - cpoint[2];
+      const auto d_sq = d0 * d0 + d1 * d1 + d2 * d2;
+      // if we have moved too much return -1 as "invalid"
+      static constexpr double factor = 1.; // factor between 0 and 1 that "invalidates cache"
```

Comment:
Shouldn't this factor be exposed to the interface? I mean it is pointless to use it always as 1.

Reply:
yes good point ---> will do. I merely introduced it here so that we have something to play with. It may even not be relevant at all.

Reply:
removed the factor for now